portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,32 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<String> messages;
```

Comment:
Use the List interface instead of the implementation type ArrayList. This way you can change the implementation (e.g. switch to a LinkedList) without having to change as much code.

Reply:
Done.

Reply:
I'm not seeing this change. Did you make it on a different branch?